src/XMakeBuildEngine/UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -2473,24 +2537,71 @@ public void PropertyFunctionStaticMethodChained()
             Assert.Equal(DateTime.Parse(_dateToParse).ToString("yyyy/MM/dd HH:mm:ss"), result);
         }
 
+#if FEATURE_SPECIAL_FOLDERS
         /// <summary>
-        /// Expand property function calls a static method an enum argument
+        /// Expand property function calls a static method available only on net46 (Environment.GetFolderPath)
         /// </summary>
         [Fact]
-        [Trait("Category", "netcore-osx-failing")]
-        public void PropertyFunctionStaticMethodEnumArgument()
+        public void PropertyFunctionGetFolderPath()
         {
             PropertyDictionary<ProjectPropertyInstance> pg = new PropertyDictionary<ProjectPropertyInstance>();
 
             Expander<ProjectPropertyInstance, ProjectItemInstance> expander = new Expander<ProjectPropertyInstance, ProjectItemInstance>(pg);
 
             string result = expander.ExpandIntoStringLeaveEscaped(@"$([System.Environment]::GetFolderPath(SpecialFolder.System))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance);
 
-#if FEATURE_SPECIAL_FOLDERS
             Assert.Equal(System.Environment.GetFolderPath(Environment.SpecialFolder.System), result);
-#else
-            Assert.Equal(FileUtilities.GetFolderPath(FileUtilities.SpecialFolder.System), result);
+        }
+#endif
+
+#if FEATURE_RUNTIMEINFORMATION
+        /// <summary>
+        /// Expand property function calls a static method available only on .netstandard1.3 and above (RuntimeInformation)
+        /// </summary>
+        [Fact]
+        public void PropertyFunctionRuntimeInformation()
+        {
+            var pg = new PropertyDictionary<ProjectPropertyInstance>();
+
+            var expander = new Expander<ProjectPropertyInstance, ProjectItemInstance>(pg);
+
+            var currentPlatformString = "";
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                currentPlatformString = OSPlatform.Windows.ToString();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                currentPlatformString = OSPlatform.Linux.ToString();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                currentPlatformString = OSPlatform.OSX.ToString();
+            }
+
+            var propertyFunction = "$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform(" +
+                                        $"$([System.Runtime.InteropServices.OSPlatform]::Create(\"{currentPlatformString}\")" +
+                                   ")))";
```

Comment:
That is a separate topic, and it should not update how this test looks because it tests a lot of stuff. I also just beefed up the test to exercise many property function cases :)